Reorder processing configuration choices

sallain opened the pull request:
I've gone through each of the processing configuration choices and reordered where necessary. Many of the changes are just for consistency - I've changed the order to `None`, `Yes`, `No` where those were the only options. For some of the jobs with more complex options I've tried to apply a more logical order:

```
Normalize:
* Normalize for preservation and access
* Normalize for preservation
* Normalize for access
* Normalize service files for access
* Normalize manually
* Do not normalize
```

```
Select compression level:
* 1 - fastest compression
* 3 - fast compression
* 5 - normal compression
* 7 - maximum compression
* 9 - ultra compression
```

```
Upload DIP:
* Upload DIP to AtoM/Binder
* Upload DIP to ArchivesSpace
* Upload DIP to CONTENTdm
* Do not upload DIP
```

One other change I've made here is in the `Generate thumbnails` option, where I've clarified the options:

```
Generate thumbnails:
* Yes
* Yes, without default icons
* No
```

The reason that this is still a WIP is because while the changes appear on the processing configuration page (Administration > Processing configuration > Default), they're not all reflected on the Transfer/Ingest tab drop-down menus. I'm not sure if this is a not-knowing-enough-about-docker problem or if there's something else in the code that I need to change.

Connected to #891
Requires https://github.com/artefactual-labs/archivematica-acceptance-tests/pull/154

sallain committed:
WIP: Reorder processing configuration choices

I've gone through each of the processing configuration choices and
reordered where necessary. Many of the changes are just for consistency
- I've changed the order to `None`, `Yes`, `No` where those were the
only options. For some of the jobs with more complex options I've tried
to apply a more logical order:

```
Normalize:
* Normalize for preservation and access
* Normalize for preservation
* Normalize for access
* Normalize service files for access
* Normalize manually
* Do not normalize
```

```
Select compression level:
* 1 - fastest compression
* 3 - fast compression
* 5 - normal compression
* 7 - maximum compression
* 9 - ultra compression
```

```
Upload DIP:
* Upload DIP to AtoM/Binder
* Upload DIP to ArchivesSpace
* Upload DIP to CONTENTdm
* Do not upload DIP
```

One other change I've made here is in the `Generate thumbnails` option,
where I've clarified the options:

```
Generate thumbnails:
* Yes
* Yes, without default icons
* No
```

The reason that this is still a WIP is because while the changes appear
on the processing configuration page (Administration > Processing
configuration > Default), they're not all reflected on the
Transfer/Ingest tab drop-down menus. I'm not sure if this is a
not-knowing-enough-about-docker problem or if there's something else in
the code that I need to change.

Connected to #891

sallain committed:
Change "Delete packages after extraction" options

Changed for consistency into `None`, `Yes`, `No` order.

ablwr commented:
TTTTHHHHHHAAAANNNKKK YOOUUUUUUU

sallain committed:
Tidy up indenting

replaceafill commented:
@sallain I think it's a problem in our code. From what I understand [the `_pull_choices` helper](https://github.com/artefactual/archivematica/blob/862c4cdd780f1a235f20716d8472d34d9b56094f/src/MCPServer/lib/RPCServer.py#L445) loses the sorting of the choices because it's returning a Python dictionary (presumably for filtering purposes?) which order is arbitrary and doesn't correspond to the chain in the workflow.

That dictionary is then [taken by the JS code](https://github.com/artefactual/archivematica/blob/4783c93467c14a95d0e4e23def141e858ac1e474/src/dashboard/src/media/js/transfer.js#L150-L168) and iterated (in the said arbitrary order) and used to build the dropdown in the UI.

sallain committed:
Make _pull_choices respect workflow ordering

With @replaceafill's help, I've made a couple small changes to force
the `_pull_choices` helper to respect the ordering of workflow choices
in workflow.json.

sallain committed:
Reorders final few items

sevein commented on src/MCPServer/lib/assets/workflow.json:
```diff
@@ -12225,6 +12213,18 @@
                         "id": "85b1e45d-8f98-4cae-8336-72f40e12cbef",
                         "items": {
                             "DeletePackage": "True"
+                      }
```
Indentation should use two extra spaces.

sevein commented on src/MCPServer/lib/server/rpc_server.py:
```diff
@@ -27,10 +27,10 @@
 from server.db import auto_close_old_connections
 from server.packages import create_package, get_approve_transfer_chain_id
 from server.processing_config import get_processing_fields
+from collections import OrderedDict
```
This import should be moved the first import block (standard library imports).

sevein commented on src/MCPServer/lib/server/rpc_server.py:
```diff
@@ -475,7 +475,9 @@ def _pull_choices(job_id, lang, jobs_awaiting_for_approval):
-    ret = {}
+
+    ret = OrderedDict()
+
```
I don't think we need the extra blank lines in L478 and L480.

sevein commented on src/MCPServer/lib/assets/workflow.json:
```diff
@@ -902,7 +902,35 @@
+                        "id": "ecfad581-b007-4612-a0e0-fcc551f4057f",
+                        "items": {
+                            "AIPCompressionLevel": "1"
+                      	}
```
Use of tab here instead of spaces, right before the closing bracket.

sevein commented on src/MCPServer/lib/assets/workflow.json:
```diff
@@ -902,7 +902,35 @@
+	                      "items": {
+	                          "AIPCompressionLevel": "3"
+	                      }
+	                  },
```
This block has indentation issues, e.g. L918 has a tab, other lines seem to be off by some spaces.

sevein reviewed (approved):
I've made some comments on cosmetics but the changes look good.

sallain committed:
Formatting fix

ablwr committed:
Set choices as OrderedDict

ablwr commented:
@replaceafill did all the work, directing me where we could simply add an `OrderedDict` to resolve this. I've tested this locally and seems well (although not extensive testing). I think this is ready for review again.

ross-spencer commented on src/MCPServer/lib/assets/workflow.json:
```diff
@@ -902,7 +902,35 @@
+                    },
+                    {
+                        "description": {
+                            "en": "5 - normal compression",
```
Unfortunately changing the option name will break AMAUAT so I feel there are two options before merging, 

* We revert the name to: `5 - normal compression mode`.
* Create a PR submitted against AMAUAT to look for the new string [here](https://github.com/artefactual-labs/archivematica-acceptance-tests/blob/14bcaef70793668d43071083ee00521077b9b862/amuser/am_browser_ability.py#L477-L480).

I have no opinion about which is the correct approach, but I'd be tempted to keep the scope of this issue to re-ordering which doesn't impact AMAUAT, and then we can address renaming another time. If there's the capacity, a PR against AMAUAT would be equally welcome now.

sallain replied:
@ross-spencer Is that the only one? I've renamed other workflow choices as well, so I'd prefer to keep those in and change the AMAUAT.

ross-spencer replied:
I believe it might be having a look today. So that's not so bad. The change needed to be made [here](https://github.com/artefactual-labs/archivematica-acceptance-tests/blob/14bcaef70793668d43071083ee00521077b9b862/amuser/am_browser_ability.py#L477-L480) and I've created a PR while I was in there so you can have a look [here](https://github.com/artefactual-labs/archivematica-acceptance-tests/pull/154). If y'all rebase and merge this during the next sprint, and then Douglas or yourself can review that change, and it will quickly show up if that's not the only problem when we run the integration tests in Jenkins again and we can fix it pretty easily. It's just something to consider for future workflow changes.

ross-spencer reviewed (commented):
UUIDs are tricky! This is nice, I've left one comment about AMAUAT which should be addressed first. Once that's done, we can get this approved and merged. Nice work all figuring it out and making the changes.

sallain commented:
@ross-spencer Coming back to this after ages and ages - your PR fixes AMAUAT, so I think this could be merged? I have just rebased.

ross-spencer commented:
@sallain nice! go for it!

@scollazo FYI! So I think we'll see the tests break, and then I can follow up and merge AMAUAT as soon as I can, and then we'll see the tests pass. And all will be well with the world again!

ross-spencer commented:
Although also, just checking @sallain right now we've seven commits associated with this PR we'd probably normally squish if possible; what approach will you use to create a single feature commit?
